### PR TITLE
Feature/trn 185/item data setter

### PR DESCRIPTION
**Pull request by oatymart**
https://oat-sa.atlassian.net/browse/TRN-185

Added `setData()` method to itemRunner API so we can use it in QTINUI provider when item data needs to be re-fetched and set during item session.

Test with `npm run test`

### CODE REVIEW

File: src/runner/api/itemRunner.js [hunk 1]
```diff
@@ -392,6 +392,19 @@ const itemRunnerFactory = function itemRunnerFactory(providerName, data = {}, op
...
         },
 
+        /**
+         * Replaces item data in rendered item
+         * @param {object} itemData
+         * @returns {itemRunner}
+         */
+        setData(itemData) {
+            data = itemData;
+            if (!closed && typeof provider.setData === 'function') {
```
Comment (olga-kulish) [inline src/runner/api/itemRunner.js]: Why is `!closed` check needed? I think 'closed' state is just a half-transparent overlay, so images will still be loaded under it.

Comment (oatymart) [reply src/runner/api/itemRunner.js]: My understanding is that `closed` is a permanent "locked" state. However, your comment makes me rethink if `setData` should still be allowed to be called here - could be a good idea if it prevents errors.

https://www.imsglobal.org/question/qtiv2p2p2/QTIv2p2p2-ASI-InformationModelv1p0/imsqtiv2p2p2_asi_v1p0_InfoModelv1p0.html#Lifecycle

File: src/runner/api/itemRunner.js [hunk 2]
```diff
@@ -392,6 +392,19 @@ const itemRunnerFactory = function itemRunnerFactory(providerName, data = {}, op
...
+                provider.setData.call(this, itemData);
+            }
+            return this;
```
Comment (krampstudio) [inline src/runner/api/itemRunner.js]: I'm wondering, in term of API, if `setData` shouldn't return a `Promise` 
It depends if we cant to either : 
1. allow `setData` to trigger a render (like your implementation in the provider)
2. or constriaints `setData` to only affect the data and then up to the provider or the consumer to "re render it" (by calling render a new time or even adding an `update` method)

if we go for 1. you should return a Promise so we will do 
```
 runner.setData(newData)
    .then( () => runner.resume() ) 
```

File: src/runner/api/itemRunner.js (hunk 1 above)
Comment (krampstudio) [reply src/runner/api/itemRunner.js]: You're correct, this is the way we have implemented it in the QTI Item Runner (closed is a final state, you can't unclose it). But we can keep the API open for any kind of implementation.
As you can see https://github.com/oat-sa/tao-item-runner-fe/pull/32/files#diff-4dee0be6c9535c2234e23b85abd2283535a62ccb5fe601d0beb1f96b0a2ab14cR480 we keep the closed state resumable from the API, not from the implementation. 
This is definitely questionable, but it should be consistent.

**Review by krampstudio — APPROVED** [review]
Looks good to me. I'll check with the other PRs

**Review by shaveko — APPROVED** [review]
Looks good, tested as a part of https://github.com/oat-sa/tao-deliver-testrunner-nui-fe/pull/91

**Comment by shaveko** [comment]
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Pull request merged**